src/cli.rs

```diff
@@ -331,11 +331,38 @@ impl Opt {
                 process::exit(1);
             }
         };
-        let root = root.unwrap_or_else(|| home.join(".sheldon"));
-        let config_file = config_file.unwrap_or_else(|| root.join("plugins.toml"));
-        let lock_file = lock_file.unwrap_or_else(|| root.join("plugins.lock"));
-        let clone_dir = clone_dir.unwrap_or_else(|| root.join("repos"));
-        let download_dir = download_dir.unwrap_or_else(|| root.join("downloads"));
+
+        let xdg_config_user = std::env::var_os("XDG_CONFIG_HOME").map(PathBuf::from);
+        let xdg_data_user = std::env::var_os("XDG_DATA_HOME").map(PathBuf::from);
+        let xdg_cache_user = std::env::var_os("XDG_CACHE_HOME");
+
+        let using_xdg =
+            xdg_data_user.is_some() || xdg_config_user.is_some() || xdg_cache_user.is_some();
```

Comment:
Do you think we should check for other vars? Like if someone has set `XDG_RUNTIME_DIR` but none of the others should we use the XDG directory structure? I initially imagined we would check for any environment variable prefixed with `XDG_` but maybe that is too broad and I realize it complicates the tests.

Reply:
I was not too sure, but I admit that checking for anything `XDG_*` is reasonable. I'll include them.

Reply:
`XDG_RUNTIME_DIR` is yet excluded from the amended commit on the basis that it is sometimes set by the system rather than the user, as is the case on e.g. [Arch Linux](`https://wiki.archlinux.org/index.php/XDG_Base_Directory). I am not sure we should interpret the variable as a requirement to use or default to the XDG specification.